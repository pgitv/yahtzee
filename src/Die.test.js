import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Die from './Die';

it('renders shallowly without crashing', function() {
  shallow(<Die handleClick={null} />);
});

it('passes snapshot test changing state to unlocked', function() {
  let wrapper = shallow(<Die handleClick={null} locked={false} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it('passes snapshot test after changing state to locked', function() {
  let wrapper = shallow(<Die handleClick={null} locked={true} />);
  //   let clicked = wrapper.find('.Die').at(0); //Check higher
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

// ``` jsx
// import React from "react";
// import { shallow } from "enzyme";
// import toJson from "enzyme-to-json";
// import GoodComponent from "./GoodComponent";

// // smoke test
// it("renders without crashing", function() {
//   shallow(<GoodComponent />);
// });

// // snapshot test
// it("matches snapshot", function() {
//   let wrapper = shallow(<GoodComponent />);
//   let serialized = toJson(wrapper);
//   expect(serialized).toMatchSnapshot();
// });
// ```
