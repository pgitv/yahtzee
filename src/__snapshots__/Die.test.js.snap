// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot test after changing state to locked 1`] = `
<button
  className="Die Die-locked"
  onClick={[Function]}
/>
`;

exports[`passes snapshot test changing state to unlocked 1`] = `
<button
  className="Die"
  onClick={[Function]}
/>
`;
